resourceLoader.js

```diff
@@ -8,124 +8,281 @@ function resourceLoader(options) {
 	'use strict';
 
 	var $ = jQuery;
-	var loader = {
+	var _loader = {
 		conditionsAllArray: []
 	};
-	var styles = [];
-	var scripts = [];
+	var _queue = [];
+
+	var _filteredQueue = [];
+	var _loadedQueue = [];
+
+	var _cache = {};
+
+	var _debug = false || options.debug;
+
+  /**
+	 * normalizes the incoming path (relative or absolute) to an absolute url
+
+	 * @param {string} path
+   * @param {object} resource
+   * @returns {string} normalizedPath
+   */
+  var normalizePath = function(path, resource) {
+    var normalizedPath;
+    var pathIsRelative = path.indexOf('http://') === -1 &&
+      path.indexOf('https://') === -1 &&
+      path.charAt(0) !== '/';
+
+    //resolve relative paths
+    if (pathIsRelative) {
+      if (resource.base) {
+        if (resource.base.substring(0, 2) === '##') {
+          resource.base = options.baseMap[resource.base];
+        }
+        path = resource.base + path;
+      } else if (options && options.base) {
+        path = options.base + path;
+      }
+    }
+    // create unique absolute path
+    normalizedPath = absolutePath(path);
+
+    (_debug) ? console.log('inputPath: ' + path) : '';
+    (_debug) ? console.log('normalizedPath: ' + normalizedPath) : '';
+    (_debug) ? console.log('------') : '';
+
+    return normalizedPath;
+  };
+
+  /**
+	 * gets all needed resources on one page and adds them to the download queue
+   */
 	var getResources = function() {
-		$.each(loader.conditionsAllArray, function () {
+    $.each(_loader.conditionsAllArray, function () {
 			$.each(this.resources, function() {
 				var resource = this;
 				var test = this.test ? this.test() : true;
 
-				if (test) {
-					$.each(resource.paths, function() {
-						var path = this;
-						var pathIsRelative = path.indexOf('http://') === -1 &&
-							path.indexOf('https://') === -1 &&
-							path.charAt(0) !== '/';
-
-						//resolve relative paths
-						if (pathIsRelative) {
-							if (resource.base) {
-								if (resource.base.substring(0, 2) === '##') {
-									resource.base = options.baseMap[resource.base];
-								}
-								path = resource.base + path;
-							} else if (options && options.base) {
-								path = options.base + path;
-							}
-						}
+        if (test) {
+          $.each(resource.paths, function() {
```

Comment:
indentation seems to be broken (tabs and spaces) can you put our default editorconfig in this repo and reformat?